Files/ViewModels/SettingsViewModel.cs

```diff
@@ -423,6 +423,21 @@ public bool AreLayoutPreferencesPerFolder
             set => Set(value);
         }
 
+        /// <summary>
+        /// Enables adaptive layout that adjusts layout mode based on the context of the directory
+        /// </summary>
+        public bool AdaptiveLayoutEnabled
+        {
+            get => Get(true);
+            set => Set(value);
+        }
+
+        public string AdaptiveLayoutPreferredLayoutMode
```

Comment:
What does this setting do?

Reply:
This setting is managed by code and user can't override it by themselves. It's purpose is to save which gridview size is preferred. I put it in settings because I had no other idea where to save it.
As always, I'm opened to suggestions

Reply:
The setting itself is fine, however, it should be together with the other layout settings, not in the FilesAndFolders region. Can you also add a description for documentation?

Reply:
@yaichenbaum I've removed this setting in https://github.com/files-community/Files/pull/3526/commits/060e41e1b5dc43d3e4058d46e9afa3e7643032b5 - different size modes are now remembered for every directory!